### ISSUE TITLE
Service does not start on modern RPM systems (CentOS 7 and Fedora 21)

### ISSUE THREAD

**Issue by chriscowley** [python-diamond/Diamond#49]
Installed an rpm from HEAD. 

Running `systemctl start diamond` gives:

```
Failed to start diamond.service: Connection timed out
```

Nothing goes into the logs

After running `diamond -f -c /etc/diamond/diamond.conf` the log contains:

```
[2015-01-03 18:35:39,751] [MainThread] Changed UID: 0 () GID: 0 ().
[2015-01-03 18:35:43,075] [MainThread] Unhandled exception: 'list' object has no attribute 'split'
[2015-01-03 18:35:43,077] [MainThread] traceback: Traceback (most recent call last):
  File "/bin/diamond", line 291, in main
    server.run()
  File "/usr/lib/python2.7/site-packages/diamond/server.py", line 96, in run
    handlers = handlers.split(',')
AttributeError: 'list' object has no attribute 'split'
```

**Comment by jaingaurav** [comment]
I believe the 'handlers' config option is supposed to be a comma separated string. Are you by chance using a list instead of a string?

**Comment by chriscowley** [comment]
I am using the defaults `handlers = diamond.handler.graphite.GraphiteHandler, diamond.handler.archive.ArchiveHandler`

I have also tried removing the space after the comma. To me that looks like a string, unless it is being enclosed in [] in server.py.

Removing `, diamond.handler.archive.ArchiveHandler` means it works, but it seems to be ignoring the host line for the GraphiteHandler. as I am not seeing any connections on the Graphite server, and the log claims `Failed to connect to localhost:2003`

**Comment by jaingaurav** [comment]
diamond 4.0 seems to have a regression in the handling of the handler string whereby it assumes that it is a string literal. ConfigObj converts any strings with a comma in it to a list.

The later issue of ignoring the GraphiteHandler block should be resolved in pull request #47 

I've also included a fix for the ConfigObj issue in the pull request

cc: @kormoc

**Comment by josegonzalez** [comment]
Closing as there was a pr opened and merged.
